styleguide/components/01-mixins/themify/_themify.scss

```diff
@@ -3,9 +3,97 @@
 //
 $themes: map-merge($themes, $overwrite-themes);
 
+//
+// Split a string based on a separator string.
+//
+// @access private
+//
+// @param $string string
+//   The string to be split.
+//
+// @param $separator string
+//   The separator by which to split the $string.
+//
+@function str-split($string, $separator: '--') {
+  // empty array/list
+  $split-arr: ();
+  // first index of separator in string
+  $index: str-index($string, $separator);
+  // loop through string
+  @while $index != null {
+    // get the substring from the first character to the separator
+    $item: str-slice($string, 1, $index - 1);
+    // push item to array
+    $split-arr: append($split-arr, $item);
+
+    @if ($separator == '--') {
+      // remove item and separator from string
+      $string: str-slice($string, $index + 2);
+    }
+    @else {
+      // remove item and separator from string
+      $string: str-slice($string, $index + 1);
+    }
+    // find new index of separator
+    $index: str-index($string, $separator);
+  }
+  // add the remaining string to list (the last item)
+  $split-arr: append($split-arr, $string);
+
+  @return $split-arr;
+}
+
+//
+// Cast a string to a number.
+//
+// @access private
+//
+// @param $value string
+//   The string to turn into a number.
+//
+@function number($value) {
+  @if type-of($value) == 'number' {
+    @return $value;
+  } @else if type-of($value) != 'string' {
```

Comment:
No need for @else if, you already returned and there is no other else option.
Just use @if.